pycarol/storage.py

```diff
@@ -22,27 +22,33 @@ def __init__(self, carol):
         """
 
         self.carol = carol
+        self.carolina = Carolina(self.carol)
+        self.backend_lock = threading.Lock()
         self._init_if_needed()
-        self.carolina = None
 
     def _init_if_needed(self):
-        # TODO: This if seems to be useless. It can be handled in `carolina.init_if_needed()`
-        if (Carolina.token is not None) and (Carolina.token.get('tenant_name', '') == self.carol.tenant['mdmName']) and \
-           (Carolina.token.get('app_name', '') == self.carol.app_name) and \
-           (datetime.utcnow() + timedelta(minutes=1) < datetime.fromtimestamp(Carolina.token.get('expirationTimestamp', 1)/1000.0)):
-            pass
-            # return
-        else:
-            Carolina.token = None
+        """
+            Check if backend has been reinitialized. If so, also reinit the GCP storage object
+
+        Returns:
+            True if storage object has been reinitialized, false if it is still valid.
+
+        """
+        if self.carolina.init_if_needed():
```

Comment:
I think the TODO comment is that you dont need to validate anything here.. If you see `StorageGCPCS` always calls `carolina.get_client()` and this is always checking if it needs to update tokens